Sangeetparashar/yld 25 stake page btc only

sangeetparashar opened the pull request:
* Revamp of stake page (mobile and desktop version)
* Added data provider to fetch prices

Sangybangy committed:
mobile view

Sangybangy committed:
lint fixes

Sangybangy committed:
add coingecko API into asset feed

Sangybangy committed:
Merge branch 'main' into sangeetparashar/yld-25-stake-page-btc-only

MattNguyen commented on src/app/components/Transactions/Transactions.tsx:
```diff
@@ -312,7 +312,7 @@ const Transactions: React.FC<{
       };
     },
-    refetchInterval: 60000, // 1 minute
+    refetchInterval: 300000, // 1 minute
```
why change this to 5 minutes?

sangeetparashar replied:
initally changed it to 5 minutes for testing, and was being throttled. Changed it back to 1 minute

sangeetparashar replied:
did not mean to make this change

sangeetparashar replied:
reverted

MattNguyen commented on src/app/components/Transactions/Transactions.tsx:
```diff
@@ -443,23 +443,31 @@ const Transactions: React.FC<{
+                      paramWithContext?.nextBlockParams?.currentVersion
+                    }
+                  />
+                  <Table.Row
```
How is this change related to the stake page?

sangeetparashar replied:
This doesnt change the stake page. I noticed spacing was incorrect in our details page and applied the fix we're using in finality provides, and all following tables

MattNguyen commented on src/app/context/AssetContext.tsx:
```diff
@@ -0,0 +1,50 @@
+// context/AssetsContext.tsx
```
Remove

MattNguyen commented on src/app/context/DataContext.tsx:
```diff
@@ -0,0 +1,58 @@
+// context/DataContext.tsx
```
remove

MattNguyen commented on src/app/stake/page.tsx:
```diff
@@ -5,103 +5,158 @@ import Image from "next/image";
+                      $0.00
+                    </div>
+                    <div className="text-yieldi-brown-light text-sm font-normal">
+                      0.0 {asset.assetSymbol}
```
do not hardcode

MattNguyen commented on src/app/stake/page.tsx:
```diff
@@ -5,103 +5,158 @@ import Image from "next/image";
+                  </Table.Cell>
+                  <Table.Cell className="px-6 py-4 ">
+                    <div className="text-yieldi-brown text-xl font-normal flex items-center h-full  ">
+                      0.00 {asset.assetSymbol}
```
do not hardcode

MattNguyen commented on src/app/config/StakedAssets.tsx:
```diff
@@ -14,6 +15,7 @@ export const assets: StakeAsset[] = [
   {
     assetName: "Native ETH",
```
Should remove eth and eth lsts until we actually integrate with eigen.

cc @kiasaki

sangeetparashar replied:
In the process of moving to .env files and will disable until we're ready for it.

MattNguyen reviewed (changes requested):
also left comments in vercel. main thing is to make sure we are using real values.

Sangybangy committed:
add .env files and map   btc wallet instead of hard coding

MattNguyen commented on src/app/stake/page.tsx:
```diff
@@ -5,103 +5,177 @@ import Image from "next/image";
+                  </Table.Cell>
+                  <Table.Cell className="px-6 py-4 items-center h-full ">
+                    <div className="text-yieldi-brown text-xl font-normal ">
+                      $0.00
```
we need to get the real tvl here

MattNguyen commented on src/app/stake/page.tsx:
```diff
@@ -5,103 +5,177 @@ import Image from "next/image";
+                    <div className="text-yieldi-brown-light text-sm font-normal">
+                      {isConnected
+                        ? (
+                            satoshiToBtc(btcWalletBalanceSat) * asset.price
```
should be tvl * asset.price

MattNguyen commented on src/app/stake/page.tsx:
```diff
@@ -5,103 +5,177 @@ import Image from "next/image";
+                    <div className="text-yieldi-brown text-xl font-normal flex items-center h-full  ">
+                      {isConnected
+                        ? (
+                            satoshiToBtc(btcWalletBalanceSat) * asset.price
```
Remove `* asset.price`. Just need the btc wallet balance

sangeetparashar replied:
done

MattNguyen committed:
chore: move /stake/btc/details to /stake/btc (#35)

Currently /stake/btc is the finality providers page for btc staking. However, this route should contain the UI that's currently on /stake/btc/details.

- Move /stake/btc to /stake/btc/providers
- Move /stake/btc/details to /stake/btc

Sangybangy committed:
update data context and its usage

Sangybangy committed:
update route

desamtralized committed:
[YLD-44][BTC staking does not result in stake position] (#36)

* refactor: Optimize delegation data management and UX

- Update Bitcoin staking success message for clarity
- Implement local storage for delegations in Transactions component
- Add logic to reconcile API and local storage delegation data
- Use combined delegations data for display, prioritizing API data
- Improve offline support and immediate feedback for delegation actions

* remove unused component & lint fix

Sangybangy committed:
formatting and metric changes

Sangybangy committed:
formatting

sangeetparashar committed:
remove dashboard from navigation/navbar (#37)

* remove dashboard from navigation/navbar

* linting

---------

Co-authored-by: Sangeet Parashar <sangeetparashar@gmail.com>

desamtralized committed:
fix: status label style and tooltip (#38)

Sangybangy committed:
map staked, pending and withdrawal balances

Sangybangy committed:
styling

Sangybangy committed:
action button updates

Sangybangy committed:
mapping delegation and staking values on finality providers

desamtralized committed:
Refactor: Update Transactions component styling and layout

- Adjust font styles and sizes in DelegationRow
- Remove 'WITHDRAWAL BALANCE' column from table
- Improve alignment and spacing in status indicator
